### DISCUSSION TITLE
Support for Gnome 42+

### DISCUSSION THREAD

**Discussion by PiotrGrobelak** [oae/gnome-shell-pano#275] [Ideas]
I have a question whether it's a technical issue that prevents full support for Gnome 42, 43...? In April, a new release of Ubuntu will come out with Gnome 46 already. Could this be developed with the mentioned support for Gnome? It's a very cool application and it would be a shame to leave it behind... If there's a possibility to expand it, I'd gladly take it on :) I'm currently using 42 myself and would be happy to help :)

**Comment by Totto16** [comment]
We already support Gnome 42-45 and are currently working on gnome 46 support.

Since in gnome 45 the whole API got a breaking change, we had to break backwards compatibility, but we still have the code [here](https://github.com/oae/gnome-shell-pano/tree/legacy). It is on a separate branch, and we will only port security and bug-fixeds to it.

If you want to help, feel free to test the extension on gnome 46 if you have a system with it. (see e.g. #265 but that's still WIP and I know already some errors and only need to fix them). Otherwise, Just lean back, and we hope to get the extension completely ready for gnome 46 until the Ubuntu 24.04 launch, since it's a big and LTS release 😉

**Comment by PiotrGrobelak** [comment]
OK, thanks for the link to the other version. I will check this when I get back home. I think it will be a nice experience for me. Actually, I don't have a system with version 46 yet, but maybe in the near future I will install the new Fedora version with GNOME 46 and check for errors, trying to provide some fixes :) I will probably be home next week and ready to work.

**Comment by PiotrGrobelak** [comment]
I'm closing this thread and asking for suggestions on what to do next.
